File: 이지수/Baekjoon_1004.swift
import Foundation

let t: Int = Int(readLine()!)!
for _ in (0..<t) {
    let di: [Int] = readLine()!.components(separatedBy: " ").map{ Int($0)! }
    let n: Int = Int(readLine()!)!
    var count: Int = 0
    for _ in (0..<n) {
        let input: [Int] = readLine()!.components(separatedBy: " ").map{ Int($0)! }
        let sin: Bool = pow(Double(input[0]-di[0]),2) + pow(Double(input[1]-di[1]),2) < pow(Double(input[2]), 2)
        let ein: Bool = pow(Double(input[0]-di[2]),2) + pow(Double(input[1]-di[3]),2) < pow(Double(input[2]), 2)
        if sin && ein { continue }
        else if sin || ein { count += 1 }
    }
    print(count)
}

File: 이지수/Python/Baekjoon_1004.py
# 1004
import sys
input = sys.stdin.readline

t = int(input())
for _ in range(t):
    x1, y1, x2, y2 = map(int, input().split())
    n = int(input())
    xs, ys, rs = [], [], []
    count = 0
    for _ in range(n) :
        x, y, r = map(int, input().split())
        sin = (x1 - x) ** 2 + (y1 - y) ** 2 < (r ** 2)
        ein = (x2 - x) ** 2 + (y2 - y) ** 2 < (r ** 2)
        if sin and ein : continue
        elif sin or ein : count += 1
    print(count)
